Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -345,59 +357,45 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
     }
 
     private func loadSelectedServer(from options: [String: AnyObject]) {
-        guard let serverName = options["selectedServer"] as? String else {
+        guard let serverName = options[NetworkProtectionOptionKey.selectedServer] as? String else {
             return
         }
 
         selectedServerStore.selectedServer = .endpoint(serverName)
     }
 
     private func loadAuthToken(from options: [String: AnyObject]) throws {
-        guard let authToken = options["authToken"] as? String else {
+        guard let authToken = options[NetworkProtectionOptionKey.authToken] as? String else {
             return
         }
 
         try tokenStore.store(authToken)
     }
 
+    private func loadExcludedRoutes(from options: [String: Any]) {
+        self.excludedRoutes = (options[NetworkProtectionOptionKey.excludedRoutes] as? [String])?.compactMap(IPAddressRange.init(from:))
+    }
+
+    // MARK: - Tunnel Start
+
     open override func startTunnel(options: [String: NSObject]?, completionHandler: @escaping (Error?) -> Void) {
         connectionStatus = .connecting
 
         // when activated by system "on-demand" the option is set
-        let isOnDemand = options?["is-on-demand"] as? Bool == true
-        let isActivatedFromSystemSettings = options?["activationAttemptId"] == nil && !isOnDemand
+        var isOnDemand: Bool {
+            options?[NetworkProtectionOptionKey.isOnDemand] as? Bool == true
+        }
+        var isActivatedFromSystemSettings: Bool {
+            options?[NetworkProtectionOptionKey.activationAttemptId] == nil && !isOnDemand
+        }
 
         let internalCompletionHandler = { [weak self] (error: Error?) in
             if error != nil {
                 self?.connectionStatus = .disconnected
-
-                // if connection is failing when activated by system on-demand
-                // ask the Main App to disable the on-demand rule to prevent activation loop
-                // To be reconsidered for the Kill Switch
-                if isOnDemand {
-                    Task { [self] in
-                        await self?.appLauncher?.launchApp(withCommand: .stopVPN)
-                        completionHandler(error)
-                    }
-                    return
-                }
```

Comment:
At first sight it looks like we won't be enabling or disabling on-demand anymore, regardless of the kill-switch status.  Is this correct?

The reason I'm asking is that I'd expect there to be no changes to on-demand when the kill switch is OFF.

Reply:
this has been moved to the macOS repo (MacPacketTunnelProvider) as mac-only feature

Reply:
Could you add a link to the lines that replace this?

Reply:
https://github.com/duckduckgo/macos-browser/blob/b5ed7355d48ac2997bf7f84b8f7fede2063cc755/DuckDuckGo/NetworkProtection/NetworkExtensionTargets/NetworkExtensionTargets/MacPacketTunnelProvider.swift#L269